tests/data/python2_unicode_literals.py

```diff
@@ -6,6 +6,7 @@
 u'hello'
 U"hello"
 Ur"hello"
+print("no spaces here please")
```

Comment:
I think this test actually passes on master! It's only when specifying `py27` as a target version that it fails

Reply:
Here's a failing test:

```diff
diff --git a/tests/data/python2_print_function.py b/tests/data/python2_print_function.py
new file mode 100755
index 0000000..c4e58c1
--- /dev/null
+++ b/tests/data/python2_print_function.py
@@ -0,0 +1,14 @@
+#!/usr/bin/env python2
+from __future__ import print_function
+
+print('hello')
+print(u'hello')
+
+# output
+
+
+#!/usr/bin/env python2
+from __future__ import print_function
+
+print("hello")
+print(u"hello")
diff --git a/tests/test_black.py b/tests/test_black.py
index 7c99f00..645eec7 100644
--- a/tests/test_black.py
+++ b/tests/test_black.py
@@ -461,6 +461,14 @@ class BlackTestCase(unittest.TestCase):
         # black.assert_equivalent(source, actual)
         black.assert_stable(source, actual, black.FileMode())
 
+    @patch("black.dump_to_file", dump_to_stderr)
+    def test_python2_print_function(self) -> None:
+        source, expected = read_data("python2_print_function")
+        mode = black.FileMode(target_versions={black.TargetVersion.PY27})
+        actual = fs(source, mode=mode)
+        self.assertFormatEqual(expected, actual)
+        black.assert_stable(source, actual, mode)
+
     @patch("black.dump_to_file", dump_to_stderr)
     def test_python2_unicode_literals(self) -> None:
         source, expected = read_data("python2_unicode_literals")
```

Reply:
opened as https://github.com/ambv/black/pull/754

Reply:
Yeah, the space is actually only a dead giveaway of the fact that Black is treating the `print` as a statement. So, add multiple tests that exercise some variants of what's possible with print:
```
print
print(a, file=sys.stderr)
print >>sys.stderr, "text"
print hey,
def print():
  ...
```

Reply:
I think most of those are already captured by tests/data/python2.py. I updated #754 with `print(a, file=sys.stderr)` though